mlir/lib/Dialect/MIOpen/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -1114,23 +1131,19 @@ struct GridwiseGemmV2RewritePattern
     int64_t MPerBlock = tuningParams.getMPerBlock();
     int64_t NPerBlock = tuningParams.getNPerBlock();
 
-    // TODO: remove in favor of new vectorization framework
-    int64_t matrix_a_source_data_per_read =
-        op->getAttr("matrix_a_source_data_per_read")
-            .template cast<IntegerAttr>()
-            .getInt();
-    int64_t matrix_b_source_data_per_read =
-        op->getAttr("matrix_b_source_data_per_read")
-            .template cast<IntegerAttr>()
-            .getInt();
-    auto matrix_a_source_vector_read_dim = static_cast<GemmDimensions>(
-        op->getAttr("matrix_a_source_vector_read_dim")
-            .template cast<IntegerAttr>()
-            .getInt());
-    auto matrix_b_source_vector_read_dim = static_cast<GemmDimensions>(
-        op->getAttr("matrix_b_source_vector_read_dim")
-            .template cast<IntegerAttr>()
-            .getInt());
+    int64_t matrix_a_source_data_per_read = 0;
+    int64_t matrix_b_source_data_per_read = 0;
+    GemmDimension matrix_a_source_vector_read_dim;
+    GemmDimension matrix_b_source_vector_read_dim;
+
+    int64_t aCopyPerThread = (KPerBlock * MPerBlock) / blockSize;
+    int64_t bCopyPerThread = (KPerBlock * NPerBlock) / blockSize;
+    GemmDimension vectorTiebreaker =
+        (KPack > 1) ? GemmDimension::K : GemmDimension::MorN;
+    std::tie(matrix_a_source_vector_read_dim, matrix_a_source_data_per_read) =
+        bestVectorization(b, op.a(), aCopyPerThread, vectorTiebreaker);
```

Comment:
I see one problem here: the xdlops path (if you look in `GemmToGridwise`) still applies a separate `kpack` dimension to the inputs. The code here doesn't account for that and so the K dimension won't be analyzed correctly.

Can you add back in `Value unkpack(Value matrix) {}` and pass the result of that to vectorization inference?

Reply:
Or, alternatively, move the function that applies kpack out of `--gemm-to-gridwise` into `--grideise-gemm-to-blockwise` and use it's result anywhere that isn't the vectorization stuff

Reply:
> Can you add back in Value unkpack(Value matrix) {} and pass the result of that to vectorization inference?

I hope I did the right thing in f5e59f4.

> Or, alternatively, move the function that applies kpack out of --gemm-to-gridwise into --grideise-gemm-to-blockwise

I figured that was part of the coordinate transforms stuff so just hold off doing that and will instead do it in next PR.

Reply:
Yeah, that's next PR